src/hdx_cli/cli_interface/user/commands.py

```diff
@@ -0,0 +1,229 @@
+from functools import partial
+import re
+import json
+import click
+
+from ...library_api.common.exceptions import LogicException
+from ...library_api.userdata.token import AuthInfo
+from ...library_api.common import rest_operations as rest_ops
+from ...library_api.utility.decorators import (report_error_and_exit,
+                                               dynamic_confirmation_prompt)
+from ...library_api.common.context import ProfileUserContext
+from ..common.undecorated_click_commands import basic_delete, basic_show, basic_create_from_dict_body
+
+
+@click.group(help="User-related operations")
+@click.option('--user', 'user_email', help='Perform operation on the passed user.',
+              metavar='USER_EMAIL', default=None)
+@click.pass_context
+def user(ctx: click.Context,
+         user_email):
+    user_profile = ctx.parent.obj['usercontext']
+    ctx.obj = {'resource_path': '/config/v1/users/',
+               'usercontext': user_profile}
+    ProfileUserContext.update_context(user_profile,
+                                      useremail=user_email)
+
+
+@click.command(help='List users.', name='list')
+@click.pass_context
+@report_error_and_exit(exctype=Exception)
+def list_(ctx: click.Context):
+    resource_path = ctx.parent.obj.get('resource_path')
+    profile = ctx.parent.obj.get('usercontext')
+    _basic_list(profile, resource_path)
+
+
+@click.command(help='Show user. If not resource_name is provided, it will show the default '
+                    'if there is one.')
+@click.option('-i', '--indent', type=int,
+              help='Number of spaces for indentation in the output.')
+@click.pass_context
+@report_error_and_exit(exctype=Exception)
+def show(ctx: click.Context, indent: int):
+    profile = ctx.parent.obj.get('usercontext')
+    resource_path = ctx.parent.obj.get('resource_path')
+    if not (resource_name := getattr(profile, 'useremail')):
+        raise LogicException('No default user found in profile.')
+    print(basic_show(profile, resource_path,
+                     resource_name,
+                     indent=indent,
+                     filter_field='email'))
+
+
+# Regular expression to validate a simple email address.
+EMAIL_REGEX = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
```

Comment:
You shouldn't need to do this validation on the client side. You can rely on the backend to validate the e-mail address. If it's not valid, the API should give an appropriate JSON response that tells you which fields failed validation.

Reply:
Agreed, I'll remove the client-side validation. Thanks!